oras/provider.py

```diff
@@ -41,6 +44,23 @@ class Subject:
     digest: str
     size: int
 
+    @classmethod
```

Comment:
I like this design! I think (now that it's more substantial) we might consider moving Subject into `oras.oci` so it's clearly organized as an OCI type. What do you think?

Reply:
That makes perfect sense to me. I'll move it. 👍